storekey-derive/src/impls/mod.rs

```diff
@@ -146,11 +164,29 @@ pub fn decode(input: TokenStream) -> Result<TokenStream> {
 
 			let mut variants = Vec::new();
 
+			let decode_type = if data_enum.variants.len() > (u8::MAX as usize) - 2 {
```

Comment:
One thing to consider here is that if you write a key that has 250 variants, then you increase to 300 variants, you will no longer be able to read the values that were already written because the code is now expecting a 16bit int instead of the 8bit value that's already on disk.

We can either document this since it's unlikely to actually be an issue that anyone runs into. Or we can just set a hard limit that values stored using storekey cannot have more than `u8::MAX - 2` variants.

Reply:
I documented it, see the change in `src/lib.rs`